pkg/sdk/metrics/metrics.go

```diff
@@ -0,0 +1,76 @@
+package metrics
+
+import (
+	"sync"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	eventsMetricName     = "operator_sdk_events"
+	syncEventsMetricName = "operator_sdk_sync_events"
+	// EventTypeLabel - metric label for event type
+	EventTypeLabel = "type"
+	// EventTypeAdd - addition event label
+	EventTypeAdd = "add"
+	// EventTypeDelete - deletion event label
+	EventTypeDelete = "delete"
+	// EventTypeUpdate - update event label
+	EventTypeUpdate = "update"
+	// SyncResultLabel - metric label for event result
+	SyncResultLabel = "result"
+	// SyncResultSuccess - successful event label
+	SyncResultSuccess = "success"
+	// SyncResultFailure - failed event label
+	SyncResultFailure = "failure"
+)
+
+var (
+	once sync.Once
+)
+
+// Collector - metric collector for all the metrics the sdk will watch
+type Collector struct {
+	Events     *prom.CounterVec
+	SyncEvents *prom.CounterVec
+}
+
+// New - create a new Collector
+func New() *Collector {
+	return &Collector{
+		Events: prom.NewCounterVec(prom.CounterOpts{
+			Name: eventsMetricName,
+			Help: "Events that the sdk has recieved, segmented by type(add or delete or update)",
+		}, []string{EventTypeLabel}),
+		SyncEvents: prom.NewCounterVec(prom.CounterOpts{
+			Name: syncEventsMetricName,
+			Help: "events that the sdk has processed segmented by result(success or failure) includes reconcilation events.",
+		}, []string{SyncResultLabel}),
+	}
+}
+
+// RegisterCollector - add collector safetly to prometheus
+func RegisterCollector(c *Collector) {
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
What is recover doing? I don't see it defined here.

Reply:
This is what recover is doing:
https://blog.golang.org/defer-panic-and-recover

I am going to change this because I forget that there is a Register method that returns an error. This will be more clear.